### PR TITLE
move controller SA to openshift-system

### PR DESCRIPTION
**Pull request by deads2k**
Let's just see if this works.

@openshift/sig-storage is there good coverage of the recycler?  This is moving it to use a different SA and namespace.  Why do you use SCC, do you really want to?

### CODE REVIEW

**Comment by deads2k** [comment]
integration and unit tests would be fairly easy to clean up.

/assign @liggitt

**Comment by deads2k** [comment]
@liggitt ready to review.  I think I fixed up the unit and integration tests.

/retest

File: vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch_recycler.go [hunk 1]
```diff
@@ -125,12 +125,22 @@ func createPVRecyclerSA(openshiftConfig string, clientBuilder controller.Control
...
+		&v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "openshift-pv-recycler",
+				Labels: map[string]string{
+					// this causes the namespace to be skipped by openshift admission plugins that require the openshift apiserver
+					"openshift.io/run-level": "1",
+				},
+			},
+		},
+	); err != nil {
```
Comment (mfojtik) [inline vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/patch_recycler.go]: missing error handling

**Comment by jsafrane** [comment]
>  is there good coverage of the recycler?

No, Origin does not run any (or very little) upstream storage tests. See https://github.com/openshift/origin/pull/18816

> This is moving it to use a different SA and namespace.

Ok,

> Why do you use SCC, do you really want to?

Heh, this is really ancient code, it has been like that since forever. What do other controllers use?

BTW, IMO it's better to postpone this to 3.11 (and enable upstream tests).

**Comment by liggitt** [comment]
metrics is moving out of openshift-infra in https://github.com/openshift/openshift-ansible/pull/8649

if that is the extent of the supported applications we run in openshift-infra, then I'm cautiously in favor of closing this PR and leaving openshift-infra without openshift-dependent admission plugins running on it.

That does require a prominent release note reiterating the nature of openshift-infra and indicating that user-deployed applications should not be placed in that namespace.

**Pull request closed without merging**